apps/src/code-studio/stageLockRedux.js

```diff
@@ -51,38 +49,33 @@ export default function reducer(state = initialState, action) {
 
   if (action.type === SET_SECTIONS) {
     const sectionId = Object.keys(action.sections)[0];
-    return Object.assign({}, state, {
-      sections: action.sections,
-      sectionsLoaded: true,
-      selectedSection: sectionId,
-    });
+    return {
+      ...state,
+      bySection: _.mapValues(action.sections, section => section.stages)
+    };
   }
 
   if (action.type === SELECT_SECTION) {
     const sectionId = action.sectionId;
-    if (!state.sections[sectionId]) {
+    if (!state.bySection[sectionId]) {
       throw new Error(`Unknown sectionId ${sectionId}`);
     }
-    const nextState = {
-      ...state,
-      selectedSection: sectionId,
-    };
-
     // If we have a lockStatus (i.e. from an open dialog) we need to update
     // it with the new section
     const { lockDialogStageId, lockStatus } = state;
```

Comment:
We're not using it (and I'll fix this). The old behavior was to always return a new object, which would set lockStatus to be the same as the current lockStatus in the case where lockDialogStageId was falsey (i.e. we were creating a new, identical object). With new changes, we'll exit out of the block and fall back to just returning 'state' in this case.